adapters/gumgum/gumgum.go

```diff
@@ -0,0 +1,145 @@
+package gumgum
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type GumGumAdapter struct {
+	URI string
+}
+
+func (g *GumGumAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	var validImps []openrtb.Imp
+	var trackingId string
+	var deviceHeight uint64
+	var deviceWidth uint64
+
+	numRequests := len(request.Imp)
+	errs := make([]error, 0, numRequests)
+
+	for i := 0; i < numRequests; i++ {
+		imp := request.Imp[i]
+		zone, devH, devW, err := preprocess(&imp)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			if request.Imp[i].Banner != nil {
+				bannerCopy := *request.Imp[i].Banner
+				if bannerCopy.W == nil && bannerCopy.H == nil && len(bannerCopy.Format) > 0 {
+					format := bannerCopy.Format[0]
+					bannerCopy.W = &(format.W)
+					bannerCopy.H = &(format.H)
+				}
+				request.Imp[i].Banner = &bannerCopy
+				validImps = append(validImps, request.Imp[i])
+				trackingId = zone
```

Comment:
You might want to do some checks if people are passing inconsistent values across the imps. Shouldn't happen in theory, but users can end up doing all sorts of crazy things.

Reply:
Is it possible for the user to send different values (unless they decide to randomly generate this value in which case none of these values would be valid on GumGum's end)? Since whatever is in the imp.ext object is hard coded when the user sets up the client for the site right?

Reply:
There is one `imp.ext` for each imp, so technically it could be possible to send inconsistent values. It would be a strong indication that there is something seriously wrong with their prebid implementation to generate a mess like that. I am not going to deny this PR if there is no check, just wanted to offer it as an idea to consider.